File: C++/StringProcessor.cpp
#include "StringProcessor.h"
#include <algorithm>

StringProcessor::StringProcessor(std::string str) {
    value = str;
}

int StringProcessor::GetLength() const {
    return value.length();
}

std::string StringProcessor::GetString() const {
    return value;
}

void StringProcessor::SortAscending() {
    std::sort(value.begin(), value.end());
}

File: C#/StringProcessor.cs
using System.Linq;

public class StringProcessor
{
    private string value;

    public StringProcessor(string str)
    {
        value = str;
    }

    public int GetLength()
    {
        return value.Length;
    }

    public string GetString()
    {
        return value;
    }

    public void SortAscending()
    {
        value = new string(value.OrderBy(c => c).ToArray());
    }
}